File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    public boolean isPrime(int number){
        if(number == 0 || number == 1){
            return false;
        }
        if(number == 2){
            return true;
        }
        for(int i = 2; i < Math.sqrt(number) + 1; i++) {
            if(number % i == 0){
                return false;
            }
        }
        return true;
    }
    
    Set<Integer> numberSet = new HashSet<>();
    boolean[] visited;
    
    public void makeNumbers(String prefix, String numbers){
        if(!prefix.equals("")){
            numberSet.add(Integer.parseInt(prefix));
        }
        
        for(int i = 0; i < numbers.length(); i++){
            if(!visited[i]){
                visited[i] = true;
                makeNumbers(prefix + numbers.charAt(i), numbers);
                visited[i] = false;
            }
        }
    }
    
    public int solution(String numbers) {
        int answer = 0;
        // int[] nums = Arrays.stream(numbers.split(""))
        //                 .mapToInt(Integer::parseInt)
        //                 .toArray();
        
        
        visited = new boolean[numbers.length()];
        makeNumbers("", numbers); // 조합 만들기
        
        for(int num : numberSet){
            if(isPrime(num)){
                //System.out.println(num);
                answer +=1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
import math

def solution(numbers):
    answer = 0
    
    def is_prime(num):
        if num == 0 or num == 1: return False
        if num == 2 : return True
            
        for i in range(2, int(math.sqrt(num))+1):
            if num % i == 0: return False
        
        return True
    
    numbers = list(numbers)
    number_set = set()
    
    for i in range(1, len(numbers)+1):
        for number in permutations(numbers,i):
            num = ''.join(number)
            number_set.add(int(num))
    
    answer = [n for n in list(number_set) if is_prime(n)]
    
    return len(answer)